server/sonar-db-dao/src/main/java/org/sonar/db/issue/IssueTesting.java

```diff
@@ -19,32 +19,59 @@
  */
 package org.sonar.db.issue;
 
+import java.util.Date;
+import org.apache.commons.lang.math.RandomUtils;
 import org.sonar.api.issue.Issue;
 import org.sonar.api.rule.Severity;
 import org.sonar.api.rules.RuleType;
 import org.sonar.api.utils.DateUtils;
 import org.sonar.core.util.Uuids;
 import org.sonar.db.component.ComponentDto;
+import org.sonar.db.rule.RuleDefinitionDto;
 import org.sonar.db.rule.RuleDto;
 
+import static com.google.common.collect.Sets.newHashSet;
+import static org.apache.commons.lang.RandomStringUtils.randomAlphabetic;
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.apache.commons.lang.math.RandomUtils.nextInt;
+
 public class IssueTesting {
 
   private IssueTesting() {
+    // only statics
+  }
 
+  public static IssueDto newIssue(RuleDefinitionDto rule, ComponentDto project, ComponentDto file) {
+    return new IssueDto()
+      .setKee("uuid_" + randomAlphabetic(5))
+      .setRule(rule)
+      .setType(RuleType.values()[nextInt(RuleType.values().length)])
+      .setProject(project)
+      .setComponent(file)
+      .setStatus(Issue.STATUS_OPEN)
+      .setResolution(null)
+      .setSeverity(Severity.ALL.get(nextInt(Severity.ALL.size())))
+      .setEffort((long) RandomUtils.nextInt(10))
+      .setAssignee("assignee_" + randomAlphabetic(5))
+      .setAuthorLogin("author_" + randomAlphabetic(5))
+      .setLine(nextInt(1_000))
+      .setMessage("message_" + randomAlphabetic(5))
+      .setChecksum("checksum_" + randomAlphabetic(5))
+      .setTags(newHashSet("tag_" + randomAlphanumeric(5), "tag_" + randomAlphanumeric(5)))
+      .setIssueCreationDate(new Date(System.currentTimeMillis() - 2_000))
+      .setIssueUpdateDate(new Date(System.currentTimeMillis() - 1_500))
+      .setCreatedAt(System.currentTimeMillis() - 1_000)
+      .setUpdatedAt(System.currentTimeMillis() - 500);
   }
 
   /**
-   * Full IssueDto used to feed database with fake data. Tests must not rely on the
-   * field contents declared here. They should override the fields they need to test,
-   * for example:
-   * <pre>
-   *   issueDao.insert(dbSession, IssueTesting.newDto(rule, file, project).setStatus(Issue.STATUS_RESOLVED).setResolution(Issue.RESOLUTION_FALSE_POSITIVE));
-   * </pre>
+   * @deprecated use newIssue(...)
    */
+  @Deprecated
   public static IssueDto newDto(RuleDto rule, ComponentDto file, ComponentDto project) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)